### DISCUSSION TITLE
Ignore non-critical schema errors?

### DISCUSSION THREAD

**Discussion by mal-tee** [composer/composer#10557] [Q&A / Support] [answered]
Hi, 

is it possible to install or update requirements from a `composer.json` even if unrelated parts of the JSON schema aren't valid?

For example, running update or install with a package like this fails:

```json
{
    "name"      : "test/Test",
    "version"   : "1.0.0",
    "require"   : {}
}
```

as the name contains uppercase chars.

Is there a way to disable composer's schema validation? I am only interested in automating the installation of potential requirements of random source code, and the `require` part of the schema seems to be valid.

**Comment by Seldaek** [comment, marked as answer]
There is no way to do this, and I'm not too keen on allowing it as really the schema should be respected otherwise it leads to other problems down the line.

**Reply by mal-tee** [reply]
Thank you for the answer. That's a completely reasonable approach. :D 

In the mean time I've resorted to falling back to composer 2, if 3 fails to verify a given `composer.json`.

**Reply by Seldaek** [reply]
I'm guessing you mean Composer 1 if 2 fails ;)

**Reply by mal-tee** [reply]
Oops. Yes :'D
